crates/sdk/src/keygen/mod.rs

```diff
@@ -155,6 +156,77 @@ where
     }
 }
 
+pub fn check_recursive_verifier_size<SC: StarkGenericConfig>(
+    vk: &MultiStarkVerifyingKey<SC>,
+    fri_params: FriParameters,
+    next_log_blowup: usize,
+) -> bool
+where
+    Val<SC>: TwoAdicField,
+{
+    let vk = &vk.inner;
+
+    // for each round we will compute the pair (total_width, num_airs, num_pts)
+    let mut rounds = vec![];
+
+    // Preprocessed round.
+    let prep_widths: Vec<usize> = vk
+        .per_air
+        .iter()
+        .filter_map(|vk| vk.params.width.preprocessed)
+        .collect();
+
+    if !prep_widths.is_empty() {
+        rounds.push((prep_widths.iter().sum::<usize>(), prep_widths.len(), 2));
+    }
+
+    let mut main_rounds = vec![];
+    for vk in vk.per_air.iter() {
+        let widths = vk.params.width.main_widths();
+        if widths.len() > main_rounds.len() {
+            main_rounds.resize(widths.len(), (0, 0, 2));
```

Comment:
I think it's weirder than this: the common main is the last main width in each vk, so you do need to check if there's a common main and if there is, pop it. (It's annoyingly complicated.) Try to just match up with the logic in the rust verifier